src/test/java/cloud/fogbow/ras/core/OrderControllerTest.java

```diff
@@ -167,36 +146,271 @@ public void testGetInactiveOrder() throws InstanceNotFoundException {
+
+        AttachmentOrder attachmentOrder1 = createLocalAttachmentOrder(computeOrder, volumeOrder1);
+        AttachmentOrder attachmentOrder2 = createLocalAttachmentOrder(computeOrder, volumeOrder2);
+
+        // exercise
+        this.ordersController.activateOrder(attachmentOrder1);
+        this.ordersController.activateOrder(attachmentOrder2);
+
+        // verify
+        Assert.assertTrue(this.orderHasDependencies(computeOrder.getId()));
+    }
+
+    // test case: Creates an order with dependencies and attempt to delete
+    // it must throw an DependencyDetectedException.
+    @Test(expected = DependencyDetectedException.class)
+    public void testDeleteOrderWithoutRemovingDependenciesFirst() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        VolumeOrder volumeOrder = createLocalVolumeOrder();
+
+        this.ordersController.activateOrder(computeOrder);
+        this.ordersController.activateOrder(volumeOrder);
+
+        AttachmentOrder attachmentOrder = createLocalAttachmentOrder(computeOrder, volumeOrder);
+        this.ordersController.activateOrder(attachmentOrder);
+
+        // exercise / verify
+        this.ordersController.deleteOrder(volumeOrder);
+    }
+
+    // test case: Creates an order with dependencies and attempt to delete
+    // them in correct order must not throw any exceptions.
+    @Test
+    public void testDeleteOrderWithDependencies() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        VolumeOrder volumeOrder = createLocalVolumeOrder();
+
+        this.ordersController.activateOrder(computeOrder);
+        this.ordersController.activateOrder(volumeOrder);
+
+        AttachmentOrder attachmentOrder = createLocalAttachmentOrder(computeOrder, volumeOrder);
+        this.ordersController.activateOrder(attachmentOrder);
+
+        // exercise / verify
+        this.ordersController.deleteOrder(attachmentOrder);
+        this.ordersController.deleteOrder(computeOrder);
+        this.ordersController.deleteOrder(volumeOrder);
+    }
+
+    // test case: Deletes an order with an instance, this should call
+    // cloud connector deleteOrder
+    @Test
+    public void testDeleteOrderWithInstanceRunning() throws FogbowException {
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        computeOrder.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        this.ordersController.activateOrder(computeOrder);
+
+        Mockito.doNothing().when(this.localCloudConnector).deleteInstance(Mockito.any());
+        
+        // exercise
+        this.ordersController.deleteOrder(computeOrder);
+
+        // verify
+        Mockito.verify(this.localCloudConnector, Mockito.times(1)).deleteInstance(Mockito.any());
     }
 
-    // test case: Checks if given an order getResourceInstance() returns its instance.
+    // test case: Deletes an order with an instance, and cloudConnector
```

Comment:
change the description to indicate that we are testing the case of forwarding the exception thrown by the cloudconnect when the order has an instance

Reply:
I'll fix the test and its description.